src/App.jsx

```diff
@@ -37,7 +38,12 @@ const App = () => {
           <Route path="/account" component={Account}></Route>
           <Route path="/register" component={Register}></Route>
           <Route path="/login" component={Login}></Route>
-          <Route path="/logout" component={Logout}></Route>
+          <Route
+            path="/logout"
+            render={() =>
+              user ? <Logout /> : <Redirect to={{ pathname: "/login" }} />
```

Comment:
In here you don't need the conditional rendering. You could just mount the Logout component all the time, and inside the Logout component you would redirect to the login, after the logout.